tensorflow/examples/tutorials/word2vec/word2vec_basic.py

```diff
@@ -102,6 +110,9 @@ def generate_batch(batch_size, num_skips, skip_window):
   span = 2 * skip_window + 1  # [ skip_window target skip_window ]
   buffer = collections.deque(maxlen=span)
   for _ in range(span):
+    if len(buffer)!=0 and data_index == 0:
+      skip_to_start(buffer, skip_window)
+      break
     buffer.append(data[data_index])
     data_index = (data_index + 1) % len(data)
```

Comment:
Note that unlike your version, which still loops around if the whole buffer is smaller than `span`, the smaller version would crash.  I think that's a reasonable policy.